src/nvim/api/private/helpers.c

```diff
@@ -760,12 +761,7 @@ bool object_to_vim(Object obj, typval_T *tv, Error *err)
     case kObjectTypeWindow:
     case kObjectTypeTabpage:
     case kObjectTypeInteger:
-      if (obj.data.integer > VARNUMBER_MAX
```

Comment:
`VARNUMBER_MAX/MIN` is defined as maximum/minimum possible value of `varnumber_T` variable. `obj.data.integer` and `varnumber_T` are both defined as `int64_t`